File: src/Scripting/PHP/patterns/unit_of_work.php
<?php declare(strict_types=1);
$check = static fn(bool $ok) => $ok ?: throw new RuntimeException('unit_of_work');
$uow = new class {
    private array $pending=[]; public array $committed=[];
    public function register(array $change): void { $this->pending[]=$change; }
    public function commit(): void { $this->committed=$this->pending; $this->pending=[]; }
};
$uow->register(['id'=>1]); $uow->commit();
$check($uow->committed === [['id'=>1]]);

File: src/Scripting/Lua/patterns/unit_of_work.lua
local store = {}
local uow = { pending = {} }
function uow:add(key, value) self.pending[key] = value end
function uow:commit()
  for key, value in pairs(self.pending) do store[key] = value end
  self.pending = {}
end
uow:add("a", 1)
assert(store.a == nil)
uow:commit()
assert(store.a == 1 and next(uow.pending) == nil)
return true
